app-api/routes/dataset.py

```diff
@@ -1293,3 +1310,381 @@ async def update_metadata(
+                if job.extra_metadata.get("type") == JobType.POLICY_SYNTHESIS.value
+            ]
+
+            # Apply cluster filter if specified
+            if cluster_name:
+                pending_jobs = [
+                    job
+                    for job in jobs_query
+                    if job.extra_metadata.get("cluster_name") == cluster_name
+                ]
+            else:
+                pending_jobs = jobs_query
+
+            # Wait a bit
+            await asyncio.sleep(0.5)
+
+            # Check if we waited too long overall
+            if (
+                datetime.datetime.now() - start_time
+            ).total_seconds() > wait_and_check_timeout:
+                break
+
+        # Return remaining jobs
+        return {
+            "jobs": [job.to_dict() for job in pending_jobs],
+        }
+
+
+@dataset.post("/jobs/cleanup")
+async def cleanup_jobs(
+    user_id: Annotated[UUID | None, Depends(UserIdentity)],
+    request: Request,
+    force: bool = False,
+    job_ids: List[str] = None,
+):
+    """
+    Administrative endpoint to clean up stale jobs.
+
+    Parameters:
+    - force: If true, clean up all jobs regardless of status
+    - job_ids: Optional list of specific job IDs to clean up
+    """
+    if job_ids:
+        # Clean up specific jobs by ID
+        with Session(db()) as session:
+            for job_id in job_ids:
+                jobs = session.query(DatasetJob).filter(DatasetJob.id == job_id).all()
+
+                for job in jobs:
+                    print(f"Manually cleaning up job {job.id}", flush=True)
+                    session.delete(job)
+
+            session.commit()
+        return {"message": f"Manually cleaned up {len(job_ids)} jobs"}
+
+    # Otherwise use the standard cleanup
+    await cleanup_stale_jobs(force_all=force)
+    return {"message": "Job cleanup complete. Check logs for details."}
+
+
+@dataset.get("/byid/{id}/generated-policies")
+async def get_generated_policies(
+    id: str,
+    user_id: Annotated[UUID | None, Depends(UserIdentity)],
+    min_detection_rate: float = 0.0,
+    success_only: bool = False,
+):
+    """
+    Get all generated policies for a dataset that have been stored in metadata.
+
+    Parameters:
+    - min_detection_rate: Minimum detection rate (0.0 to 1.0) to include policy
+    - success_only: If True, only include policies where success=True
+    """
+    # refresh job statuses when client requests this endpoint (e.g. to process potentially
+    # completed jobs, since the last check)
+    await check_all_jobs()
+
+    with Session(db()) as session:
+        # Check if the user has access to the dataset
+        dataset = session.query(Dataset).filter(Dataset.id == id).first()
+        if not dataset:
+            raise HTTPException(status_code=404, detail="Dataset not found")
+        if dataset.user_id != user_id and not dataset.is_public:
+            raise HTTPException(status_code=403, detail="Access denied")
```

Comment:
We should not allow non-owners to see the generated policies. Non-owners won't see the Guardrails tab in the UI, right now. 

Here this means, we should not allow GET when the dataset is is_public, unless the requester owns it.